db/kvstore4trie.go

```diff
@@ -83,8 +88,21 @@ func (s *KVStoreForTrie) Delete(key []byte) error {
 	return nil
 }
 
+// Purge marks a key for future deletion when the trie is to be pruned
+func (s *KVStoreForTrie) Purge(tag, key []byte) error {
+	trieKeystoreMtc.WithLabelValues("purge").Inc()
+	// tag will be used as a criterion to determine if the key should be deleted
+	// it is simply prepended in front of the key and stored into the Prune namespace
+	// it is up to the caller to define the exact format of tag and how to use it
+	k := make([]byte, len(tag)+len(key))
+	copy(k, tag)
+	k = append(k, key...)
```

Comment:
append k means append data from index len(tag)+len(key),this should be k := make([]byte, len(tag)) or make([]byte,0) and append(k,tag)

Reply:
good catch, fixed in latest patch